cli/cli-debug/src/test/java/io/telicent/smart/cache/cli/commands/debug/DockerTestDebugCliMutualTlsKafka.java

```diff
@@ -0,0 +1,199 @@
+        }
+
+        // Uncomment for easier debugging in IDE
+        //SmartCacheCommandTester.TEE_TO_ORIGINAL_STREAMS = true;
+        super.setup();
+        setupLogging();
+        this.kafka.setup();
+    }
+
+    private void setupLogging() {
+        enableSpecificLogging(KafkaEventSource.class, Level.DEBUG);
+    }
+
+    private void teardownLogging() {
+        enableSpecificLogging(KafkaEventSource.class, Level.OFF);
+    }
+
+
+    @AfterMethod
+    @Override
+    public void testCleanup() throws InterruptedException {
+        super.testCleanup();
+        this.kafka.resetTestTopic();
+
+        // Occasionally we can get random test errors because the test topic(s) don't get recreated in time which can
+        // lead to incomplete test data being generated in the subsequent test.  A short sleep makes that unlikely to
+        // occur.
+        Thread.sleep(500);
+    }
+
+    @AfterClass
+    @Override
+    public void teardown() {
+        this.kafka.teardown();
+        teardownLogging();
+        super.teardown();
+    }
+
+    private void generateKafkaEvents(String format) {
+        generateKafkaEvents(Collections.emptyList(), format);
+    }
+
+    private void generateKafkaEvents(Collection<Header> headers, String format) {
+        try (KafkaSink<String, String> sink = KafkaSink.<String, String>create()
+                                                       .keySerializer(StringSerializer.class)
+                                                       .valueSerializer(StringSerializer.class)
+                                                       .bootstrapServers(this.kafka.getBootstrapServers())
+                                                       .topic(KafkaTestCluster.DEFAULT_TOPIC)
+                                                       .producerConfig(this.kafka.getClientProperties())
+                                                       .lingerMs(5)
+                                                       .build()) {
+            for (int i = 1; i <= 1_000; i++) {
+                sink.send(new SimpleEvent<>(headers, Integer.toString(i),
+                                            String.format(format, i)));
+            }
+        }
+    }
+
+    @Test
+    public void givenEmptyTopic_whenDumpingEvents_thenNothingDumped() {
+        // Given and When
+        DebugCli.main(new String[] {
+                "dump",
+                "--bootstrap-servers",
+                this.kafka.getBootstrapServers(),
+                "--topic",
+                KafkaTestCluster.DEFAULT_TOPIC,
+                "--kafka-properties",
+                CLIENT_PROPERTIES_FILE,
+                "--max-stalls",
+                "1",
+                "--poll-timeout",
+                "3",
+                "--read-policy",
+                "BEGINNING",
+                "--no-live-reporter",
+                "--no-health-probes"
+        });
```

Comment:
I understand this could be too picky, but the command arguments seem to be repeated across tests, if these change in future is could be worth considering exporting the common CLI arguments into a reusable methods.

E.G
```java
private String[] buildCommonDumpArgs(String command) {
    return new String[] {
        command,
        "--bootstrap-servers", this.kafka.getBootstrapServers(),
        "--topic", KafkaTestCluster.DEFAULT_TOPIC,
        "--kafka-properties", CLIENT_PROPERTIES_FILE,
        "--max-stalls", "1",
        "--poll-timeout", "5",
        "--read-policy", "BEGINNING",
        "--no-live-reporter",
        "--no-health-probes"
    };
}

@Test
public void whenNonEmptyTopic_thenEventsDumpedSuccessfully() {
    DebugCli.main(buildCommonDumpArgs("dump"));
}
```